railties/test/generators/plugin_generator_test.rb

```diff
@@ -148,6 +148,32 @@ def test_generating_adds_dummy_app_without_sprockets
     end
   end
 
+  def test_generator_for_yarn
```

Comment:
How about letting specify the application path from each test? As described below. 

```diff 
diff --git a/railties/test/generators/app_generator_test.rb b/railties/test/generators/app_generator_test.rb
index ccb437c..cb46000 100644
--- a/railties/test/generators/app_generator_test.rb
+++ b/railties/test/generators/app_generator_test.rb
+
+    def application_path
+      destination_root
+    end
 end
diff --git a/railties/test/generators/plugin_generator_test.rb b/railties/test/generators/plugin_generator_test.rb
index bd76af8..f55b89a 100644
--- a/railties/test/generators/plugin_generator_test.rb
+++ b/railties/test/generators/plugin_generator_test.rb
@@ -150,6 +150,18 @@ def test_generating_adds_dummy_app_without_sprockets
     end
   end
 
+  private
+    def application_path
+      "#{destination_root}/test/dummy"
+    end
 end
diff --git a/railties/test/generators/shared_generator_tests.rb b/railties/test/generators/shared_generator_tests.rb
index c73b91e..d584fbb 100644
--- a/railties/test/generators/shared_generator_tests.rb
+++ b/railties/test/generators/shared_generator_tests.rb
@@ -114,6 +114,13 @@ def test_skip_git
     assert_no_directory(".git")
   end
 
+  def test_generator_for_yarn
+    run_generator([destination_root])
+
+    assert_file "#{application_path}/package.json", /dependencies/
+    assert_file "#{application_path}/config/initializers/assets.rb", /node_modules/
+  end
```